plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/fs/nio/EFSNIOFolder.java

```diff
@@ -66,7 +79,14 @@ public void createLink(URI location, int updateFlags, IProgressMonitor monitor)
 
     public void delete(boolean force, boolean keepHistory, IProgressMonitor monitor) throws CoreException {
         try {
-            Files.delete(getNioPath());
+            if (getRoot().getRoot().getFileSystem().supportsEmptyFolders()) {
+                Files.delete(getNioPath());
+            } else {
+                // We can't delete pseudo-folder, so this is a workaround
```

Comment:
Looks dangerous and unsafe.
Can we delete only our FOLDER_EMULATION_FILE_NAME?

Reply:
Then folder will not be deleted on delete action